rundeckapp/grails-spa/packages/ui-trellis/src/library/components/storage/tests/KeyStoragePage.spec.ts

```diff
@@ -0,0 +1,171 @@
+import { flushPromises, mount, VueWrapper } from "@vue/test-utils";
+import { ComponentPublicInstance } from "vue";
+import KeyStoragePage from "../KeyStoragePage.vue";
+import { getRundeckContext } from "../../../rundeckService";
+import { Modal } from "uiv";
+jest.mock("../../../rundeckService", () => ({
+  getRundeckContext: jest.fn().mockReturnValue({
+    rdBase: "mockRdBase",
+    projectName: "test-project",
+    rundeckClient: {
+      storageKeyGetMetadata: jest.fn().mockResolvedValue({
+        resources: [
+          {
+            name: "newKey",
+            path: "/keys/newKey",
+            type: "file",
+            meta: { rundeckKeyType: "private" },
+          },
+        ],
+        _response: { status: 200 },
+      }),
+      storageKeyDelete: jest.fn().mockResolvedValue({}),
+      projectList: jest.fn().mockResolvedValue([]),
+    },
+  }),
+  url: jest.fn().mockReturnValue({ href: "mockHref" }),
+}));
+interface KeyStoragePageData {
+  modalEdit: boolean;
+  selectedKey: Record<string, any>;
+}
+type KeyStoragePageComponent = ComponentPublicInstance & KeyStoragePageData;
+const mountKeyStoragePage = async (props = {}) => {
+  return mount(KeyStoragePage as unknown as KeyStoragePageComponent, {
+    props: {
+      project: "test-project",
+      readOnly: true,
+      allowUpload: true,
+      modelValue: "",
+      storageFilter: "",
+      ...props,
+    },
+    global: {
+      components: {
+        Modal,
+      },
+      mocks: {
+        $t: (msg: string) => msg,
+      },
+    },
+  }) as unknown as VueWrapper<KeyStoragePageComponent>;
+};
+
+describe("KeyStoragePage.vue", () => {
+  let rundeckClientMock: any;
+  beforeEach(() => {
+    rundeckClientMock = getRundeckContext().rundeckClient;
```

Comment:
nit: the rundeckClientMock doesn't need to be reassigned every time. Would move this line to 55 and keep it as a const;